__tests__/app.test.js

```diff
@@ -71,39 +74,72 @@ describe("/api/articles", () => {
-          expect(response.body.article.article_img_url).toBe(
-            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
-          );
+        .then(({ body }) => {
+          const { article } = body;
+          expect(article.article_id).toBe(3);
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.body).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
         });
     });
 
     test("GET 404: Responds with an error when passed a non-existent id", () => {
       return request(app)
         .get("/api/articles/99999")
         .expect(404)
-        .then((response) => {
-          expect(response.body.message).toBe("Article not found.");
+        .then(({ body }) => {
+          expect(body.message).toBe("Article not found.");
         });
     });
 
     test("GET 400: Responds with an error when passed an id of an invalid data type", () => {
       return request(app)
         .get("/api/articles/invalid-id")
         .expect(400)
-        .then((response) => {
-          expect(response.body.message).toBe("Bad request.");
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+  });
+
+  describe("GET /api/articles/:article_id/comments", () => {
+    test("GET 200: Responds with an array with all comment objects of a particular article with the required key-value pairs ordered from most recent", () => {
+      return request(app)
+        .get("/api/articles/5/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments.length).toBe(2);
+          comments.forEach((comment) => {
+            expect(comment.article_id).toBe(5);
+            expect(typeof comment.comment_id).toBe("number");
+            expect(typeof comment.votes).toBe("number");
+            expect(typeof comment.created_at).toBe("string");
+            expect(typeof comment.author).toBe("string");
+            expect(typeof comment.body).toBe("string");
+          });
+        });
+    });
+
+    test("GET 404: Responds with an error when passed a non-existent id", () => {
+      return request(app)
+        .get("/api/articles/99999/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("Article not found.");
+        });
+    });
+
+    test("GET 400: Responds with an error when passed an id of an invalid data type", () => {
+      return request(app)
+        .get("/api/articles/invalid-id/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
         });
```

Comment:
Excellent error testing here! Specifically, your descriptions explain exactly what I need to expect from the endpoint, well done.